### ISSUE TITLE
Youtube Channel Video Scroll Limit

### ISSUE THREAD

**Issue by BackThrowTheBaguetteBrawler** [omega-gg/MotionBox#27]
As of right now, you cannot scroll through an entire channel's library, only the first 30 or so videos. This makes it harder to find older videos.

**Comment by itsserial** [comment]
I concur, such a feature would really strengthen the application.

**Comment by JayBeeDe** [comment]
I have a YouTube playlist with lot of videos, only the first 100 videos are shown... @3unjee Is there something I missed ?
Here is the playlist URL (to test): https://www.youtube.com/playlist?list=PLxEgtk5k-NNhpkp5bLM_CfiZhUQZM-jVY

**Comment by Tharn** [comment]
Agreed, this is huge. If you only get 30 videos when looking at a channel, MotionBox will always be just a YT player and not a proper YT browser. With playlists, channels and subscription options already present in MotionBox you would expect the application to be "complete". Please improve this, it will make MotionBox so much better.

**Comment by orobas0** [comment]
I agree as well. To be able to retrieve all the videos from a channel instead of the last 30 would be a game changer, it's the most important thing I miss. Browsing the full video list would probably feel clumsier with so many items, considering that some channels have thousands of videos... but pagination (like the one present in "Related videos") + sorting and/or filtering buttons would solve that.

I'll keep using MotionBox hoping for improvements on that front. Its simplicity, responsiveness and low memory footprint are fantastic.

**Comment by HOAMTH** [comment]
Just in case anyone is searching for a solution, I have at least a workaround:
Download a list of all the songs in your youtube playlist using [yt-dlp](https://github.com/yt-dlp/yt-dlp). Download yt-dlp.exe and open cmd. Assuming you kept the yt-dlp.exe in your Downloads folder, put the following into cmd:
`yt-dlp.exe --flat-playlist --get-url https://www.youtube.com/playlist?list=<YourPlaylistID> > playlist.txt`
This should create a text file with all urls in the same folder where the yt-dlp.exe is. Open Motionbox and click on the adress bar. Type in the following: `file:///Users/<YourWindowsUsername>/Downloads/playlist.txt` (or whatever the path to the playlist.txt is). Motionbox will then fetch every single song from that list of URLs. Now simply select all of them with Ctrl+A and and either add them to an existing playlist of create a new playlist.
